src/python_testing/TC_DRLK_2_9.py

```diff
@@ -0,0 +1,566 @@
+                     "Verify that the DUT sends SUCCESS response."),
+            TestStep("15a", "TH reads NumberOfCredentialsSupportedPerUser attribute from DUT.",
+                     "Verify that TH is able to read the attribute successfully."),
+            TestStep("15b", "TH sends SetUser Command to DUT.",
+                     "Verify that the DUT sends SUCCESS response."),
+            TestStep("15c", "TH sends SetCredential Command to DUT.",
+                     "Verify that the DUT sends SetCredentialResponse command  with Status SUCCESS."),
+            TestStep("15d", " TH sends SetCredential Command  with CredentialIndex as 'max-num-credential-user'.",
+                     "Verify that the DUT sends SetCredentialResponse command with Status RESOURCE_EXHAUSTION."),
+            TestStep("15e", "TH sends ClearCredential Command to DUT to clear all the credentials of PIN type.",
+                     "Verify that the DUT sends SUCCESS response."),
+            TestStep("16", " TH sends SetUser Command to DUT.",
+                     "Verify that the DUT sends SUCCESS response"),
+            TestStep("17", "TH sends SetCredential Command to DUT with CredentialType.RFID.",
+                     "Verify that the DUT sends SetCredentialResponse command  with Status SUCCESS."),
+            TestStep("18", "TH sends SetCredential Command to DUT with CredentialType.PIN.",
+                     "Verify that the DUT sends SetCredentialResponse command  with Status SUCCESS."),
+            TestStep("19", "TH sends GetUser Command to DUT.",
+                     "Verify that the DUT responds with GetUserResponse Command."),
+            TestStep("20", "TH sends SetCredential Command to DUT to modify the CredentialData.",
+                     "Verify that the DUT sends SetCredentialResponse command  with Status SUCCESS."),
+            TestStep("21", "TH sends GetUser Command to DUT.",
+                     "Verify that the DUT responds with GetUserResponse Command."),
+            TestStep("22", "TH sends ClearCredential Command to DUT to clear all the credentials.",
+                     "Verify that the DUT sends SUCCESS response."),
+            TestStep("23", "TH sends ClearUser Command to DUT.",
+                     "Verify that the DUT sends SUCCESS response."),
+        ]
+        return steps
+
+    async def read_attributes_from_dut(self, endpoint, cluster, attribute, expected_status: Status = Status.Success):
+        try:
+            attribute_value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster,
+                                                                             attribute=attribute)
+            asserts.assert_equal(expected_status, Status.Success)
+            return attribute_value
+        except Exception as e:
+            logging.error(e)
+            asserts.assert_equal(expected_status, Status.Success,
```

Comment:
in the except block replace Status.success with any other forms like Status.InvalidState ot Status.constraintError etc since Status.Success will not be produced when excpet blocks are reached.